Dealing with extra C includes

What is the correct way to deal with any C files that are included with a project. Currently I am trying to switch `vector` to use `doctest-parallel`. For `doctest` setup was:

```haskell
import Test.DocTest (doctest)

main :: IO ()
main = doctest ["-Iinclude", "-Iinternal", "src/Data"]
```

but for `doctest-parallel` suggested setup:

```haskell
main :: IO ()
main = mainFromCabal "vector" =<< getArgs
```

doesn't work and yields errors along the lines:

```
/home/lehins/haskell/vector/vector/src/Data/Vector/Fusion/Bundle/Monadic.hs:117:2: error:
     fatal error: vector.h: No such file or directory
      117 | #include "MachDeps.h"
          |  ^~~~~~~~~~
    |
117 | #include "MachDeps.h"
    |  ^
compilation terminated.

/home/lehins/haskell/vector/vector/src/Data/Vector/Fusion/Bundle.hs:109:2: error:
     fatal error: vector.h: No such file or directory
      109 | 
          |  ^         
    |
109 | 
    |  ^
compilation terminated.
```

Any suggestions? If it is possible to make it work then it would be great to add an example to README. It is not uncommon for users to have cbits for performance critical parts.

--- a/CHANGES.markdown
+++ b/CHANGES.markdown
@@ -1,3 +1,6 @@
+# 0.2.1
+  * C include directories (Cabal field: `include-dirs`) are now passed to GHC when parsing source files ([#7](https://github.com/martijnbastiaan/doctest-parallel/issues/7))
+
 # 0.2
 Changes:
   * Support for GHC 9.2 has been added ([#4](https://github.com/martijnbastiaan/doctest-parallel/pull/4))

--- a/doctest-parallel.cabal
+++ b/doctest-parallel.cabal
@@ -118,6 +118,8 @@ library spectests-modules
   -- ghc-options: -Wall
   hs-source-dirs:
       test/integration
+  include-dirs:
+      test/integration/WithCInclude/include
   c-sources:
       test/integration/WithCbits/foo.c
   exposed-modules:
@@ -157,6 +159,7 @@ library spectests-modules
     TestSimple.Fib
     TrailingWhitespace.Foo
     WithCbits.Bar
+    WithCInclude.Bar
 
 test-suite spectests
   main-is: Spec.hs

--- a/src/Test/DocTest/Helpers.hs
+++ b/src/Test/DocTest/Helpers.hs
@@ -1,6 +1,5 @@
 {-# LANGUAGE CPP #-}
 {-# LANGUAGE MultiWayIf #-}
-{-# LANGUAGE NamedFieldPuns #-}
 {-# LANGUAGE RecordWildCards #-}
 {-# LANGUAGE LambdaCase #-}
 
@@ -25,7 +24,7 @@ import Distribution.Types.UnqualComponentName ( unUnqualComponentName )
 import Distribution.PackageDescription
   ( CondTree(CondNode, condTreeData), GenericPackageDescription (condLibrary)
   , exposedModules, libBuildInfo, hsSourceDirs, defaultExtensions, package
-  , packageDescription, condSubLibraries )
+  , packageDescription, condSubLibraries, includeDirs )
 import Distribution.Pretty (prettyShow)
 import Distribution.Verbosity (silent)
 
@@ -38,16 +37,22 @@ import Distribution.PackageDescription.Parsec (readGenericPackageDescription)
 
 data Library = Library
   { libSourceDirectories :: [FilePath]
+    -- ^ Haskell source directories
+  , libCSourceDirectories :: [FilePath]
+    -- ^ C source directories
   , libModules :: [ModuleName]
+    -- ^ Exposed modules
   , libDefaultExtensions :: [Extension]
+    -- ^ Extensions enabled by default
   }
   deriving (Show)
 
 -- | Convert a "Library" to arguments suitable to be passed to GHCi.
 libraryToGhciArgs :: Library -> ([String], [String], [String])
-libraryToGhciArgs Library{..} = (srcArgs, modArgs, extArgs)
+libraryToGhciArgs Library{..} = (hsSrcArgs <> cSrcArgs, modArgs, extArgs)
  where
-  srcArgs = map ("-i" <>) libSourceDirectories
+  hsSrcArgs = map ("-i" <>) libSourceDirectories
+  cSrcArgs = map ("-I" <>) libCSourceDirectories
   modArgs = map prettyShow libModules
   extArgs = map showExt libDefaultExtensions
 
@@ -144,10 +149,12 @@ extractSpecificCabalLibrary maybeLibName pkgPath = do
     let
       buildInfo = libBuildInfo lib
       sourceDirs = hsSourceDirs buildInfo
+      cSourceDirs = includeDirs buildInfo
       root = takeDirectory pkgPath
     in
       pure Library
         { libSourceDirectories = map ((root </>) . compatPrettyShow) sourceDirs
+        , libCSourceDirectories = map (root </>) cSourceDirs
         , libModules = exposedModules lib
         , libDefaultExtensions = defaultExtensions buildInfo
         }

--- a/test/MainSpec.hs
+++ b/test/MainSpec.hs
@@ -160,3 +160,7 @@ spec = do
     it "doesn't get confused by doctests using System.IO imports" $ do
       doctest ["SystemIoImported.A"]
         (cases 2)
+
+    it "correctly handles C import directories" $ do
+      doctest ["WithCInclude.Bar"]
+        (cases 1)

--- a/test/integration/WithCInclude/Bar.hs
+++ b/test/integration/WithCInclude/Bar.hs
@@ -1,0 +1,12 @@
+{-# LANGUAGE CPP #-}
+
+module WithCInclude.Bar where
+
+#include "WithCInclude.h"
+
+
+-- |
+-- >>> x
+-- 42
+x :: Int
+x = THE_DEFINE

--- a/test/integration/WithCInclude/include/WithCInclude.h
+++ b/test/integration/WithCInclude/include/WithCInclude.h
@@ -1,0 +1,1 @@
+#define THE_DEFINE 42

